### PR TITLE
Refactor the new version of the core

### PR DESCRIPTION
**Pull request by Elyas-Amini**
### Checklist:

- [ ] It refers to an [Issue](https://github.com/EasyMicroservices/FileManager/issues).
- [ ] It fixes a bug.
- [x] It's a new feature (non-breaking change which adds functionality)
- [ ] It's a breaking (fix or feature that would cause existing functionality to change)
- [ ] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] CHANGELOG is updated.
- [ ] Related language label is selected.

### CODE REVIEW

**Review by MahdiyarGHD — APPROVED** [review]

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Database/Database/Contexts/SupportContext.cs [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 
+        public DbSet<DepartmentEntity> Departmants { get; set; }
+        public DbSet<TicketAssignEntity> TicketAssigns { get; set; }
+        public DbSet<TicketDepartmentEntity> TicketDepartments { get; set; }
         public DbSet<TicketEntity> Tickets { get; set; }
-        public DbSet<TicketHistoryEntity> TicketHistory { get; set; }
-        public DbSet<TicketDepartmentEntity> TicketDepartment { get; set; }
-
+        public DbSet<TicketHistoryEntity> TicketHistories { get; set; }
+        public DbSet<TicketSupportTimeHistoryEntity> TicketSupportTimeHistoryHistories { get; set; }
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Database/Database/Contexts/SupportContext.cs]:
```suggestion
        public DbSet<TicketSupportTimeHistoryEntity> TicketSupportTimeHistories { get; set; }
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Database/EasyMicroservices.SupportsMicroservice.Database.csproj [hunk 2]
```diff
@@ -11,7 +11,9 @@
   
   <ItemGroup>
 	  <PackageReference Include="EasyMicroservices.Configuration" Version="0.0.0.2" />
-	  <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.3" />
+	  <PackageReference Include="EasyMicroservices.Cores.AspCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Database/EasyMicroservices.SupportsMicroservice.Database.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Database/EasyMicroservices.SupportsMicroservice.Database.csproj [hunk 3]
```diff
@@ -11,7 +11,9 @@
...
+	  <PackageReference Include="EasyMicroservices.Cores.AspEntityFrameworkCoreApi" Version="0.0.0.49" />
+	  <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Database/EasyMicroservices.SupportsMicroservice.Database.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Database/EasyMicroservices.SupportsMicroservice.Database.csproj [hunk 4]
```diff
@@ -11,7 +11,9 @@
   
   <ItemGroup>
 	  <PackageReference Include="EasyMicroservices.Configuration" Version="0.0.0.2" />
-	  <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.3" />
+	  <PackageReference Include="EasyMicroservices.Cores.AspCoreApi" Version="0.0.0.49" />
+	  <PackageReference Include="EasyMicroservices.Cores.AspEntityFrameworkCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Database/EasyMicroservices.SupportsMicroservice.Database.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj [hunk 5]
```diff
@@ -11,7 +11,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="EasyMicroservices.Cores.Database" Version="0.0.0.15" />
+    <PackageReference Include="EasyMicroservices.Cores.AspCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj [hunk 6]
```diff
@@ -11,7 +11,10 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.AspEntityFrameworkCoreApi" Version="0.0.0.49" />
+    <PackageReference Include="EasyMicroservices.Cores.Database" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj [hunk 7]
```diff
@@ -11,7 +11,10 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj [hunk 8]
```diff
@@ -6,9 +6,12 @@
   </PropertyGroup>
   
   <ItemGroup>
-    <PackageReference Include="EasyMicroservices.Mapper.CompileTimeMapper" Version="0.0.0.5" />
-    <PackageReference Include="EasyMicroservices.Mapper.SerializerMapper" Version="0.0.0.2" />
-    <PackageReference Include="EasyMicroservices.Serialization.Newtonsoft.Json" Version="0.0.0.3" />
+    <PackageReference Include="EasyMicroservices.Cores.AspCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj [hunk 9]
```diff
@@ -6,9 +6,12 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.AspEntityFrameworkCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj [hunk 10]
```diff
@@ -6,9 +6,12 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj [hunk 11]
```diff
@@ -6,9 +6,12 @@
...
+    <PackageReference Include="EasyMicroservices.Mapper.CompileTimeMapper" Version="0.0.0.6" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Logics/EasyMicroservices.SupportsMicroservice.Logics.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Services/EasyMicroservices.SupportsMicroservice.Services.csproj [hunk 12]
```diff
@@ -4,6 +4,12 @@
     <TargetFramework>net7.0</TargetFramework>
     <RootNamespace>EasyMicroservices.SupportsMicroservice</RootNamespace>
   </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="EasyMicroservices.Cores.AspCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Services/EasyMicroservices.SupportsMicroservice.Services.csproj:9]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Services/EasyMicroservices.SupportsMicroservice.Services.csproj [hunk 13]
```diff
@@ -4,6 +4,12 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.AspEntityFrameworkCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Services/EasyMicroservices.SupportsMicroservice.Services.csproj:10]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Services/EasyMicroservices.SupportsMicroservice.Services.csproj [hunk 14]
```diff
@@ -4,6 +4,12 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Services/EasyMicroservices.SupportsMicroservice.Services.csproj:11]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/DatabaseBuilder.cs [hunk 15]
```diff
@@ -9,16 +10,18 @@
...
+        public DatabaseBuilder(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
 
         public void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
         {
-            optionsBuilder.UseInMemoryDatabase("SupportDatabase");
-            //optionsBuilder.UseSqlServer(config.GetConnectionString("local"));
+            //optionsBuilder.UseInMemoryDatabase("CommentDatabase");
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/DatabaseBuilder.cs]:
```suggestion
            //optionsBuilder.UseInMemoryDatabase("SupportDatabase");
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj [hunk 16]
```diff
@@ -6,7 +6,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="EasyMicroservices.WhiteLabelsMicroservice.Clients" Version="0.0.0.6" />
+    <PackageReference Include="EasyMicroservices.Cores.AspCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj [hunk 17]
```diff
@@ -6,7 +6,10 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.AspEntityFrameworkCoreApi" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj [hunk 18]
```diff
@@ -6,7 +6,10 @@
...
+    <PackageReference Include="EasyMicroservices.Cores.Relational.EntityFrameworkCore" Version="0.0.0.49" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj [hunk 19]
```diff
@@ -6,7 +6,10 @@
...
+    <PackageReference Include="EasyMicroservices.WhiteLabelsMicroservice.Clients" Version="0.0.0.9" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.StartUp/EasyMicroservices.SupportsMicroservice.StartUp.csproj]:
```suggestion
```

File: src/CSharp/EasyMicroservices.SupportsMicroservice.WebApi/Controllers/DepartmentController.cs [hunk 20]
```diff
@@ -3,14 +3,21 @@
...
+
+
 
 namespace EasyMicroservices.SupportsMicroservice.WebApi.Controllers
 {
     public class DepartmentController : SimpleQueryServiceController<DepartmentEntity, CreateDepartmentRequestContract, UpdateDepartmentRequestContract, DepartmentContract, long>
     {
-        public DepartmentController(IContractLogic<DepartmentEntity, CreateDepartmentRequestContract, UpdateDepartmentRequestContract, DepartmentContract, long> contractLogic) : base(contractLogic)
-        {
+        private readonly IContractLogic<DepartmentEntity, CreateDepartmentRequestContract, UpdateDepartmentRequestContract, DepartmentContract, long> _contractlogic;
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.WebApi/Controllers/DepartmentController.cs]: You don't need to use contractlogic property. use unitofwork inside of your methods

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]

**Review by mahdidelzendeh72 — APPROVED** [review]
thanks

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj [hunk 21]
```diff
@@ -6,15 +6,12 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <Folder Include="Contracts\Requests\Filters\" />
+    <Folder Include="Contracts  Requests\Filters\" />
```
Comment (Ali-YousefiTelori) [inline src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj]: its wrong path

**Review by Ali-YousefiTelori — CHANGES_REQUESTED** [review]

File: src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj (hunk 21 above)
Comment (Elyas-Amini) [reply src/CSharp/EasyMicroservices.SupportsMicroservice.Domain/EasyMicroservices.SupportsMicroservice.Domain.csproj]: It was fixed

**Review by Ali-YousefiTelori — APPROVED** [review]

**Pull request merged**